homeassistant/components/number/__init__.py

```diff
@@ -247,7 +263,7 @@ def device_class(self) -> NumberDeviceClass | None:
             return self.entity_description.device_class
```

Comment:
Would the cache be invalidated if the `entity_description` added later? I don't think we allow changing `entity_description`, because it is by itself cached in the entity registry, but hat about adding it late?

Reply:
https://github.com/home-assistant/core/pull/106271#discussion_r1435384746